db/document.go

```diff
@@ -829,6 +837,48 @@ func (doc *Document) UpdateExpiry(expiry uint32) {
 
 //////// CHANNELS & ACCESS:
 
+func (doc *Document) updateChannelHistory(channelName string, seq uint64, addition bool) {
+	// Check if we already have an entry for
+	for idx, historyEntry := range doc.ChannelHistory {
+		if historyEntry.Name == channelName {
+			// If we are here there is an existing entry for this channel
+			// If addition we need to:
+			// - Move existing history entry to old channels
+			// - Remove existing entry from current channels
+			// - Add new entry with start seq
+			// If removal / not addition then we can simply add the end seq
+			if addition {
+				doc.OldChannelHistory = append(doc.OldChannelHistory, historyEntry)
```

Comment:
Think about whether we should be defensive here and not write items to OldChannelHistory if historyEntry.end isn't present.

Reply:
I'd be very surprised if this happens as we're plugging into the existing channel addition / removal, however, it doesn't hurt to be defensive.
Added a check and if there is no existing 'end' when attempting to do an addition we're going to just return. This means we just keep the existing record with the existing start and no end. In terms of how this could affect revocation I believe worse case is we end up sending a revocation for a doc ID which the user has never had. This isn't perfect but without the information of when a removal was carried out our options are limited.